File: LATIHAN4DPBO2023/CPP/Prodi.cpp
/*Saya Mochamad Khaairi NIM 2106416 mengerjakan Latihan 4
dalam mata kuliah Desain dan Pemrograman Berorientasi Objek 
untuk keberkahanNya maka saya tidak melakukan kecurangan 
seperti yang telah dispesifikasikan. Aamiin.*/

//include file header
#include "header.hh"

//inisialisasi constructor
Prodi::Prodi()
{
	namaProdi = "";
	kode = "";
}

//inisialisasi constructor dengan parameter
Prodi::Prodi(string namaProdi, string kode, Mahasiswa mahasiswa, Dosen dosen, Course course)
{
	this->namaProdi = namaProdi;
	this->kode = kode;
	this->mahasiswa.push_back(mahasiswa);
	this->dosen.push_back(dosen);
	this->courses.push_back(course);
}

//inisialisasi method setter dan getter
void Prodi::setNamaProdi(string namaProdi)
{
	this->namaProdi = namaProdi;
}

string Prodi::getNamaProdi()
{
	return this->namaProdi;
}


void Prodi::setKode(string kode)
{
	this->kode = kode;
}

string Prodi::getKode()
{
	return this->kode;
}

void Prodi::setMahasiswa(Mahasiswa mahasiswa)
{
	this->mahasiswa.push_back(mahasiswa);
}

list<Mahasiswa> Prodi::getMahasiswa()
{
	return this->mahasiswa;
}

void Prodi::setDosen(Dosen dosen)
{
	this->dosen.push_back(dosen);
}

list<Dosen> Prodi::getDosen()
{
	return this->dosen;
}

void Prodi::setCourses(Course course)
{
	this->courses.push_back(course);
}

list<Course> Prodi::getCourses()
{
	return this->courses;
}

//inisialisasi destructor
Prodi::~Prodi()
{
}

File: LATIHAN4DPBO2023/Python/Prodi.py
# inisialisasi class Prodi
class Prodi:
	__namaProdi = ""
	__kode = ""
	__courses = []
	__mahasiswa = []
	__dosen = []

	# constructor
	# untuk courses, mahasiswa, dosen akan diinput dengan object, composite dari class Mahasiswa, class Dosen dan class Course
	def __init__(self, course, mahasiswa, dosen, namaProdi = "", kode = ""):
		self.__namaProdi = namaProdi
		self.__kode = kode
		# karena berbentuk list, maka diinputnya menggunakan append
		# nantinya list akan berisi kumpulan object
		self.__courses.append(course)
		self.__mahasiswa.append(mahasiswa)
		self.__dosen.append(dosen)

	# kumpulan method setter dan getter untuk tiap atribut
	def setNamaProdi(self, namaProdi):
		self.__namaProdi = namaProdi

	def getNamaProdi(self):
		return self.__namaProdi

	def setKode(self, kode):
		self.__kode = kode

	def getKode(self):
		return self.__kode

	def setCourses(self, course):
		self.__courses.append(course)

	def getCourses(self):
		return self.__courses

	def setMahasiswa(self, mahasiswa):
		self.__mahasiswa.append(mahasiswa)

	def getMahasiswa(self):
		return self.__mahasiswa

	def setDosen(self, dosen):
		self.__dosen.append(dosen)

	def getDosen(self):
		return self.__dosen